noir-projects/aztec-nr/easy-private-state/src/easy_private_uint.nr

```diff
@@ -24,19 +24,28 @@ impl<Context> EasyPrivateUint<Context> {
 impl<Context> EasyPrivateUint<&mut PrivateContext> {
     // Very similar to `value_note::utils::increment`.
     pub fn add(self, addend: u64, owner: AztecAddress, outgoing_viewer: AztecAddress) {
-        let owner_npk_m_hash = get_current_public_keys(self.context, owner).npk_m.hash();
+        let owner_keys = get_current_public_keys(self.context, owner);
+        let outgoing_viewer_keys = get_current_public_keys(self.context, outgoing_viewer);
         // Creates new note for the owner.
-        let mut addend_note = ValueNote::new(addend as Field, owner_npk_m_hash);
+        let mut addend_note = ValueNote::new(addend as Field, owner_keys.npk_m.hash());
 
         // Insert the new note to the owner's set of notes.
         // docs:start:insert
-        self.set.insert(&mut addend_note).emit(encode_and_encrypt_note(self.context, outgoing_viewer, owner));
+        self.set.insert(&mut addend_note).emit(
+            encode_and_encrypt_note_with_keys(
+                self.context,
+                outgoing_viewer_keys.ovpk_m,
+                owner_keys.ivpk_m,
+                owner
```

Comment:
Do you recall why we still pass the 'recipient' here?

Reply:
The recipient travels all the way to [EncryptedLogOutgoingBody](https://github.com/AztecProtocol/aztec-packages/blob/5bff26b2f9aecb8298225d5abe72740fedd1f4e8/noir-projects/aztec-nr/aztec/src/encrypted_logs/outgoing_body.nr#L31) where it's part of the encrypted log. Then the EncryptedLogOutgoingBody gets decrypted by NoteProcessor [here](https://github.com/AztecProtocol/aztec-packages/blob/5bff26b2f9aecb8298225d5abe72740fedd1f4e8/yarn-project/circuit-types/src/logs/l1_payload/l1_payload.ts#L143). The recipient is currently completely unused there but that might be because we have not yet fully build the outgoing support in PXE. We store the outgoing notes as [OutgoingNoteDaos](https://github.com/AztecProtocol/aztec-packages/blob/5bff26b2f9aecb8298225d5abe72740fedd1f4e8/yarn-project/pxe/src/database/outgoing_note_dao.ts#L10) and those don't really have the info of who was the "recipient".

Looking at this more it really is quite suspicious because the outgoingBody contains recipientIvpk and you could always derive recipient from that (e.g. by checking all the txs interacting with the key registry contract for or smt.).

@nventuro do you have more context on this or shall we summon Mike?

Reply:
> recipientIvpk and you could always derive recipient from that

Assuming the keys have ever been registered, or assuming we have a derivation scheme in which this backwards derivation is possible, neither of which need be true. Even then I'm not entirely sure why you'd care about the recipient.

It seems this was created in https://github.com/AztecProtocol/aztec-packages/pull/7390 due to https://github.com/AztecProtocol/aztec-packages/issues/7177, but I'm not sure _why_. We'll be looking at logs soon and this will come up then I imagine.